gitlab/resource_gitlab_project_variable.go

```diff
@@ -0,0 +1,102 @@
+package gitlab
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabProjectVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabProjectVariableCreate,
+		Read:   resourceGitlabProjectVariableRead,
+		Update: resourceGitlabProjectVariableUpdate,
+		Delete: resourceGitlabProjectVariableDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
It might be worth adding a `ValidateFunc` here to give plan time failures if they don't match the requirements set out in https://docs.gitlab.com/ee/api/project_level_variables.html#create-variable
> The key of a variable; must have no more than 255 characters; only A-Z, a-z, 0-9, and _ are allowed